### PR TITLE
[SDPA-4029] revokes permission

### PR DESCRIPTION
**Pull request by vincent-gao**
## JIRA
https://digital-engagement.atlassian.net/browse/SDPA-4029

## CHANGES
1. delete `administer taxonomy` from yml files.
2. add `access taxonomy overview` to yml files.
3. an hook_update for revoking  `administer taxonomy`  and granting `access taxonomy overview` from drupal site
4.  approver and site_admin both can still edit or create new terms.

https://github.com/dpc-sdp/content-vic-gov-au/pull/847

### CODE REVIEW

**Comment by anthony-malkoun** [comment]
> @vincent-gao My reading of the issue is that delete access should be removed from the Sites vocab only. This doesn't look like it's targeting the sites vocab specifically but all vocabs.

I think that's right @GROwen as then the normal permissions will take effect. The permission being granted is the equivalent of `view content` for taxonomy terms.

File: tide_core.install [hunk 1]
```diff
@@ -442,3 +443,25 @@ function tide_core_update_8018() {
...
+ *
+ * Revokes 'administer taxonomy' permission from site_admin and approver roles,
+ * and grants them 'access taxonomy overview'.
+ */
+function tide_core_update_8019() {
+  $roles = ['approver', 'site_admin'];
+  $taxonomy_permissions = new TaxonomyPermissions(\Drupal::entityManager());
+  foreach ($roles as $role_id) {
+    $assigned_permissions = ['access taxonomy overview'];
+    foreach ($taxonomy_permissions->permissions() as $permission => $details) {
```
Comment (anthony-malkoun) [inline tide_core.install:458]: What are you doing here? Are you removing the `delete` permission?

**Comment by anthony-malkoun** [comment]
> > @vincent-gao My reading of the issue is that delete access should be removed from the Sites vocab only. This doesn't look like it's targeting the sites vocab specifically but all vocabs.
> 
> I think that's right @GROwen as then the normal permissions will take effect. The permission being granted is the equivalent of `view content` for taxonomy terms.

Actually I think @GROwen is right (I seem to be making a habit of this)... I don't think you should be trying to modify existing permissions beyond doing what you need to do.

**Comment by GROwen** [comment]
@vincent-gao With the opening of dpc-sdp/tide_site#53 can this PR be closed?

**Comment by vincent-gao** [comment]
yes. @GROwen . this one should be closed.

**Pull request closed without merging**